nomos-blend/proofs/src/selection/mod.rs

```diff
@@ -44,11 +44,9 @@ impl ProofOfSelection {
     pub fn expected_index(&self, membership_size: usize) -> Result<usize, Error> {
         // Condition 1: https://www.notion.so/nomos-tech/Blend-Protocol-215261aa09df81ae8857d71066a80084?source=copy_link#215261aa09df819991e6f9455ff7ec92
         let selection_randomness_bytes = fr_to_bytes(&self.selection_randomness);
-        let selection_randomness_blake_hash =
-            blake2b512(&[&DOMAIN_SEPARATION_TAG[..], &selection_randomness_bytes[..]]);
         let pseudo_random_output: u64 = {
             let pseudo_random_output_bytes =
-                pseudo_random_sized_bytes::<8>(&selection_randomness_blake_hash);
+                pseudo_random_sized_bytes::<8>(&DOMAIN_SEPARATION_TAG, &selection_randomness_bytes);
```

Comment:
Mhh I guess I misunderstood what `pseudo_random_sized_bytes` does then? The [spec](https://www.notion.so/nomos-tech/Blend-Protocol-215261aa09df81ae8857d71066a80084?source=copy_link#215261aa09df8110a059d0eb477559b5) defines PoSel as the modulo of the pseudo-random bytes over the domain-separated selection randomness **hash** and the membership size. Why are we not hashing here anymore? Is the hashing performed internally? If that's the case, how do we generate pseudo-random bytes without necessarily hashing?

Reply:
Hashing is performed internally. I understand why this could be confusing. The spec defines hash functions and CSPRBG separately, but in the code, we've integrated them because we always have to hash the input for CSPRBG. We may be able to try to refactor this to match with the spec, but I didn't want to change it in this PR.

Reply:
I updated the doc comments to clearly mention that the hashing is done internally. 6a94b99f

Reply:
and I think this design is actually ergonomic and prevent misuses.